core/worker.go

```diff
@@ -1710,6 +1671,200 @@ func (w *worker) prepareWork(genParams *generateParams, wo *types.WorkObject) (*
 		}
 	}
 
+	if nodeCtx == common.PRIME_CTX {
+		newWo.Header().SetPrimeStateRoot(types.EmptyRootHash)
+	}
+	if nodeCtx == common.REGION_CTX {
+		newWo.Header().SetRegionStateRoot(types.EmptyRootHash)
+	}
+	// compute and set the miner difficulty
+	if nodeCtx == common.PRIME_CTX {
```

Comment:
If we are computing this in the working we neeed to be checking in the state processor.